SIGSEGV when agent is not found

```
cagent run --agent unknown --yolo ./examples/gopher.yaml
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x98 pc=0x1057c52bc]

goroutine 1 [running]:
github.com/docker/cagent/pkg/agent.(*Agent).MaxIterations(...)
	/Users/dgageot/src/cagent/pkg/agent/agent.go:65
github.com/docker/cagent/cmd/root.doRunCommand({0x1061246a0, 0x14000521a00}, {0x140005263c0, 0x1, 0x4}, 0x0)
	/Users/dgageot/src/cagent/cmd/root/run.go:282 +0xc6c
github.com/docker/cagent/cmd/root.runCommand(0x140005b8608, {0x140005263c0, 0x1, 0x4})
	/Users/dgageot/src/cagent/cmd/root/run.go:83 +0x64
github.com/spf13/cobra.(*Command).execute(0x140005b8608, {0x14000526380, 0x4, 0x4})
	/Users/dgageot/go/pkg/mod/github.com/spf13/cobra@v1.10.1/command.go:1015 +0x7d4
github.com/spf13/cobra.(*Command).ExecuteC(0x140005b8008)
	/Users/dgageot/go/pkg/mod/github.com/spf13/cobra@v1.10.1/command.go:1148 +0x350
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/dgageot/go/pkg/mod/github.com/spf13/cobra@v1.10.1/command.go:1071
github.com/spf13/cobra.(*Command).ExecuteContext(...)
	/Users/dgageot/go/pkg/mod/github.com/spf13/cobra@v1.10.1/command.go:1064
github.com/docker/cagent/cmd/root.Execute()
	/Users/dgageot/src/cagent/cmd/root/root.go:152 +0x1d0
main.main()
	/Users/dgageot/src/cagent/main.go:10 +0x1c
```

--- a/cmd/root/debug.go
+++ b/cmd/root/debug.go
@@ -36,7 +36,12 @@ func debugToolsetsCommand(cmd *cobra.Command, args []string) error {
 	}
 
 	for _, name := range team.AgentNames() {
-		agent := team.Agent(name)
+		agent, err := team.Agent(name)
+		if err != nil {
+			slog.Error("Failed to get agent", "name", name, "error", err)
+			continue
+		}
+
 		slog.Info("Query tools", "name", agent.Name())
 		tools, err := agent.Tools(ctx)
 		if err != nil {

--- a/cmd/root/run.go
+++ b/cmd/root/run.go
@@ -278,8 +278,13 @@ func doRunCommand(ctx context.Context, args []string, exec bool) error {
 		rt = remoteRt
 		slog.Debug("Using remote runtime", "address", remoteAddress, "agent", agentName)
 	} else {
+		agent, err := agents.Agent(agentName)
+		if err != nil {
+			return err
+		}
+
 		// Create session first to get its ID for OAuth state encoding
-		sess = session.New(session.WithMaxIterations(agents.Agent(agentName).MaxIterations()))
+		sess = session.New(session.WithMaxIterations(agent.MaxIterations()))
 		sess.ToolsApproved = autoApprove
 
 		// Create local runtime with root session ID for OAuth state encoding
@@ -817,10 +822,12 @@ func getCommandsForAgent(agentFilename string, isRemote bool, agents *team.Team,
 		if agents == nil {
 			return nil, fmt.Errorf("failed to load agent team")
 		}
-		ag := agents.Agent(agentName)
-		if ag == nil {
-			return nil, fmt.Errorf("agent not found: %s", agentName)
+
+		ag, err := agents.Agent(agentName)
+		if err != nil {
+			return nil, err
 		}
+
 		return ag.Commands(), nil
 	}
 

--- a/pkg/runtime/runtime.go
+++ b/pkg/runtime/runtime.go
@@ -165,15 +165,8 @@ func New(agents *team.Team, opts ...Opt) (Runtime, error) {
 	}
 
 	// Validate that we have at least one agent and that the current agent exists
-	if r.team == nil || r.team.Size() == 0 {
-		return nil, fmt.Errorf("no agents loaded; ensure your agent configuration defines at least one agent")
-	}
-	if r.team.Agent(r.currentAgent) == nil {
-		names := strings.Join(r.team.AgentNames(), ", ")
-		if names == "" {
-			names = "(none)"
-		}
-		return nil, fmt.Errorf("agent %q not found in team; available agents: %s", r.currentAgent, names)
+	if _, err = r.team.Agent(r.currentAgent); err != nil {
+		return nil, err
 	}
 
 	slog.Debug("Creating new runtime", "agent", r.currentAgent, "available_agents", agents.Size())
@@ -182,7 +175,9 @@ func New(agents *team.Team, opts ...Opt) (Runtime, error) {
 }
 
 func (r *runtime) CurrentAgent() *agent.Agent {
-	return r.team.Agent(r.currentAgent)
+	// We validated already that the agent exists
+	current, _ := r.team.Agent(r.currentAgent)
+	return current
 }
 
 func (r *runtime) StopPendingProcesses(ctx context.Context) error {
@@ -222,13 +217,12 @@ func (r *runtime) RunStream(ctx context.Context, sess *session.Session) <-chan E
 		))
 		defer sessionSpan.End()
 
-		a := r.team.Agent(r.currentAgent)
-
 		// Set the events channel for elicitation requests
 		r.setElicitationEventsChannel(events)
 		defer r.clearElicitationEventsChannel()
 
 		// Set elicitation handler on all MCP toolsets before getting tools
+		a := r.CurrentAgent()
 		for _, toolset := range a.ToolSets() {
 			toolset.SetElicitationHandler(r.elicitationHandler)
 			toolset.SetOAuthSuccessHandler(func() {
@@ -928,15 +922,15 @@ func (r *runtime) handleTaskTransfer(ctx context.Context, sess *session.Session,
 
 	slog.Debug("Creating new session with parent session", "parent_session_id", sess.ID, "tools_approved", sess.ToolsApproved)
 
-	subAgentMaxIter := 0
-	if child := r.team.Agent(params.Agent); child != nil {
-		subAgentMaxIter = child.MaxIterations()
+	child, err := r.team.Agent(params.Agent)
+	if err != nil {
+		return nil, err
 	}
 
 	s := session.New(
 		session.WithSystemMessage(memberAgentTask),
 		session.WithImplicitUserMessage("", "Follow the default instructions"),
-		session.WithMaxIterations(subAgentMaxIter),
+		session.WithMaxIterations(child.MaxIterations()),
 	)
 	s.SendUserMessage = false
 	s.Title = "Transferred task"

--- a/pkg/runtime/runtime_test.go
+++ b/pkg/runtime/runtime_test.go
@@ -531,9 +531,7 @@ func TestNewRuntime_InvalidCurrentAgentError(t *testing.T) {
 
 	// Ask for a non-existent current agent
 	_, err := New(tm, WithCurrentAgent("other"), WithModelStore(mockModelStore{}))
-	require.Error(t, err)
-	require.Contains(t, err.Error(), "agent \"other\" not found")
-	require.Contains(t, err.Error(), "root") // available agents listed in error
+	require.Contains(t, err.Error(), "agent not found: other (available agents: root)")
 }
 
 func TestProcessToolCalls_UnknownTool_NoToolResultMessage(t *testing.T) {

--- a/pkg/server/server.go
+++ b/pkg/server/server.go
@@ -539,11 +539,17 @@ func (s *Server) importAgent(c echo.Context) error {
 
 	s.teams[agentKey] = t
 
+	rootAgent, err := t.Agent("root")
+	if err != nil {
+		slog.Error("Failed to get root agent", "error", err)
+		return jsonErr(c, http.StatusInternalServerError, "failed to get root agent: "+err.Error())
+	}
+
 	slog.Info("Agent imported successfully", "originalPath", validatedSourcePath, "targetPath", targetPath, "key", agentKey)
 	return c.JSON(http.StatusOK, map[string]string{
 		"originalPath": validatedSourcePath,
 		"targetPath":   targetPath,
-		"description":  t.Agent("root").Description(),
+		"description":  rootAgent.Description(),
 	})
 }
 
@@ -659,7 +665,13 @@ func (s *Server) pullAgent(c echo.Context) error {
 
 	s.teams[agentName] = t
 
-	return c.JSON(http.StatusOK, map[string]string{"name": agentName, "description": t.Agent("root").Description()})
+	rootAgent, err := t.Agent("root")
+	if err != nil {
+		slog.Error("Failed to get root agent", "error", err)
+		return jsonErr(c, http.StatusInternalServerError, "failed to get root agent: "+err.Error())
+	}
+
+	return c.JSON(http.StatusOK, map[string]string{"name": agentName, "description": rootAgent.Description()})
 }
 
 func (s *Server) pushAgent(c echo.Context) error {
@@ -782,8 +794,8 @@ func (s *Server) getAgents(c echo.Context) error {
 
 	var agents []api.Agent
 	for id, t := range s.teams {
-		a := t.Agent("root")
-		if a == nil {
+		a, err := t.Agent("root")
+		if err != nil {
 			slog.Error("Agent root not found", "team", id)
 			continue
 		}
@@ -1033,8 +1045,9 @@ func (s *Server) runAgent(c echo.Context) error {
 	if err != nil {
 		return jsonErr(c, http.StatusInternalServerError, "failed to load agent for session")
 	}
-	agent := t.Agent(currentAgent)
-	if agent == nil {
+
+	agent, err := t.Agent(currentAgent)
+	if err != nil {
 		return jsonErr(c, http.StatusNotFound, fmt.Sprintf("agent not found: %s", currentAgent))
 	}
 

--- a/pkg/team/team.go
+++ b/pkg/team/team.go
@@ -2,7 +2,9 @@ package team
 
 import (
 	"context"
+	"errors"
 	"fmt"
+	"strings"
 
 	"github.com/docker/cagent/pkg/agent"
 )
@@ -44,14 +46,22 @@ func (t *Team) AgentNames() []string {
 	return names
 }
 
-func (t *Team) Agent(name string) *agent.Agent {
-	if t.agents == nil {
-		return nil
+func (t *Team) Agent(name string) (*agent.Agent, error) {
+	if len(t.agents) == 0 {
+		return nil, errors.New("no agents loaded; ensure your agent configuration defines at least one agent")
 	}
-	if _, ok := t.agents[name]; !ok {
-		return nil
+
+	found, ok := t.agents[name]
+	if !ok {
+		var names []string
+		for n := range t.agents {
+			names = append(names, n)
+		}
+
+		return nil, fmt.Errorf("agent not found: %s (available agents: %s)", name, strings.Join(names, ", "))
 	}
-	return t.agents[name]
+
+	return found, nil
 }
 
 func (t *Team) Size() int {

